src/NuGet.Services.AzureSearch/Catalog2AzureSearch/CatalogIndexActionBuilder.cs

```diff
@@ -199,48 +219,91 @@ private async Task<IndexChanges> GetIndexChangesAsync(Context context)
         private IndexAction<KeyedDocument> GetSearchIndexAction(
             Context context,
             SearchFilters searchFilters,
-            SearchIndexChangeType changeType)
+            SearchIndexChangeType changeType,
+            string[] owners)
         {
             var latestFlags = _search.LatestFlagsOrNull(context.VersionLists, searchFilters);
             Guard.Assert(
                 changeType == SearchIndexChangeType.Delete || latestFlags != null,
                 "Either the search document is being or there is a latest version.");
 
-            switch (changeType)
+            if (changeType == SearchIndexChangeType.Delete)
             {
-                case SearchIndexChangeType.Delete:
-                    return IndexAction.Delete(_search.Keyed(
-                        context.PackageId,
-                        searchFilters));
-
-                case SearchIndexChangeType.UpdateVersionList:
+                return IndexAction.Delete(_search.Keyed(
+                    context.PackageId,
+                    searchFilters));
+            }
+            else if (changeType == SearchIndexChangeType.UpdateVersionList)
+            {
+                if (owners != null)
```

Comment:
Do you think this code would be cleaner if, regardless of owners, we always called the same method on `_search`, but that method determined what to do with the owners?

Reply:
Spoke offline. Will keep as-is to be consistent with the other `if` forks.